File: 0775-global-and-local-inversions/0775-global-and-local-inversions.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var isIdealPermutation = function(nums) {
    // let local = 0;
    // let global = 0;
    let max = -1
    for(let i=0; i < nums.length -2; i++){
        // if(nums[i] > nums[i+1])local++;
        // exNums = nums.slice(i+2, nums.length)
        // if(exNums.length >= 1 && exNums.some(el => el < nums[i])){
        //     return false
        // }
        max = Math.max(nums[i], max)
        if(max > nums[i+2]) return false
            
    }
    return true;

};

File: 0775-global-and-local-inversions/0775-global-and-local-inversions.py
class Solution:
    def isIdealPermutation(self, nums: List[int]) -> bool:
        maxNum = -1
        i = 0
        while i < len(nums) -2:
            # maxNum = max(nums[i], maxNum) 
            if nums[i] >= maxNum:
                maxNum = nums[i]
            if maxNum > nums[i+2]:
                return False
            i += 1
        return True